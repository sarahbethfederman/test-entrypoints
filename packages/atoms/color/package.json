{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  