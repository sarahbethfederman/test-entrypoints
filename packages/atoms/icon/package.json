{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
