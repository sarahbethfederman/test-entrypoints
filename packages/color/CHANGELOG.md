2.0.2 - New Changelog
