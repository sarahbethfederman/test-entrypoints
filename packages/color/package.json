{
  "name": "@lendi-ui/color",
  "version": "3.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^6.0.0",
    "styled-components": "^3"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^2.0.0"
  },
  "scripts": {
    "build:esm": "tsc --project tsconfig.esm.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:esm"
  }
}
