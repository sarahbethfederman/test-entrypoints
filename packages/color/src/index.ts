import { css } from 'styled-components';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type NameOrNameMap = BreakpointValue<string> | BreakpointValueMap<string>;

export function color(name: NameOrNameMap) {
  return select(`colors.${name}`);
}

export function fg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            color: ${color(n)};
          `
  );
}

export function bg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            background-color: ${color(n)};
          `
  );
}
