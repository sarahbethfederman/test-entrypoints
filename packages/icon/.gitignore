src/icons-compiled/*
src/index.ts