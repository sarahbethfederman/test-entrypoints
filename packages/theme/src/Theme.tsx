import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
