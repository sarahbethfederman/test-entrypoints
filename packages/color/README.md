# @lendi-ui/color

Color style with `bg`(background color), `fg`(font color) and `color`(every CSS property relate to color).

## Installation

```
yarn add @lendi-ui/color
```

## Usage

- `bg` for styling background-color.

```jsx
import { bg } from '@lendi-ui/color';
const Square = styled.div`
  width: 2em;
  height: 2em;
  ${bg('warn.500')};
`;
```

- `fg` for styling font-color.

```jsx
import { fg } from '@lendi-ui/color';
const WarningText = styled.span`
  ${fg('warn.500')};
`;
```

- `color` for styling every CSS porperty relate to color.

```jsx
import { color } from '@lendi-ui/color';
const Circle = styled.div`
  width: 0;
  border-radius: 1em;
  border: 1em solid ${color('primary.500')};
`;
```
