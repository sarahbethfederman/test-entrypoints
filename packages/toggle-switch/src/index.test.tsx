import * as React from 'react';
import { mount } from 'enzyme';
import Theme, { theme } from '@lendi-ui/theme';
import { ErrorOutline } from '@lendi-ui/icon';
import Spinner from '@lendi-ui/spinner';
import { color } from '@lendi-ui/color';
import ToggleSwitch from './index';
import { ToggleLabel, Wrapper } from './index.style';
import { IconWrapper, ToggleCheckbox, ToggleHandle, ToggleTrack, ToggleWrapper } from './Toggle/index.style';

let element;
let mockOnChange: jest.Mock<{}>;

function render(props) {
  mockOnChange = jest.fn();

  element = mount(
    <Theme>
      <ToggleSwitch value={'1'} label={'Test Toggle Switch'} onChange={mockOnChange} {...props} />
    </Theme>
  );
}

describe('ToggleSwitch', () => {
  it('should render the whole component', () => {
    render({});
    expect(element.find(ToggleSwitch)).toHaveLength(1);
    expect(element.find(ToggleCheckbox)).toHaveLength(1);
    expect(element.find(ToggleHandle)).toHaveLength(1);
    expect(element.find(ToggleLabel)).toHaveLength(1);
    expect(element.find(ToggleTrack)).toHaveLength(1);
    expect(element.find(ToggleWrapper)).toHaveLength(1);
    expect(element.find(Wrapper)).toHaveLength(1);

    // Snapshot
    expect(element.find(ToggleSwitch)).toMatchSnapshot();
  });

  it('should have the correct checked property based on passed props', () => {
    render({ isChecked: false });
    expect(element.find(ToggleCheckbox).props().checked).toBeFalsy();

    render({ isChecked: true });
    expect(element.find(ToggleCheckbox).props().checked).toBeTruthy();
  });

  it('should render correct default styles when off', () => {
    render({});

    expect(element.find(ToggleTrack)).toHaveStyleRule('background-color', color('shade.300')({ theme }));
    expect(element.find(ToggleHandle)).toHaveStyleRule('background-color', color('shade.25')({ theme }));
  });

  it('should render correct default styles when on', () => {
    render({ isChecked: true });

    expect(element.find(ToggleTrack)).toHaveStyleRule('background-color', color('primary.300')({ theme }));
    expect(element.find(ToggleHandle)).toHaveStyleRule('background-color', color('primary.500')({ theme }));
  });

  it('should render correct styles when disabled', () => {
    render({ isDisabled: true });

    const testee = element.find(Wrapper);
    expect(testee).toHaveStyleRule('cursor', 'not-allowed');
    expect(testee).toHaveStyleRule('pointer-events', 'none');
    expect(testee).toHaveStyleRule('opacity', '0.4');
  });

  it('should render the correct icon when loading', () => {
    render({ isLoading: true });

    expect(element.find(IconWrapper)).toHaveLength(1);
    expect(element.find(Spinner)).toHaveLength(1);
  });

  it('should render the correct colours and icon when error', () => {
    render({ isError: true });

    expect(element.find(ToggleTrack)).toHaveStyleRule('background-color', color('error.200')({ theme }));
    expect(element.find(ToggleHandle)).toHaveStyleRule('background-color', color('error.500')({ theme }));

    expect(element.find(IconWrapper)).toHaveLength(1);
    expect(element.find(ErrorOutline)).toHaveLength(1);
  });

  it('should render the correct styles when boxed and off', () => {
    render({ isBoxed: true });

    const testee = element.find(Wrapper);
    expect(testee).toHaveStyleRule('border', `1px solid ${color('shade.200')({ theme })}`);
  });

  it('should render the correct styles when boxed and on', () => {
    render({ isBoxed: true, isChecked: true });

    const testee = element.find(Wrapper);
    expect(testee).toHaveStyleRule('background-color', color('primary.50')({ theme }));
    expect(testee).toHaveStyleRule('border', `1px solid ${color('primary.500')({ theme })}`);
  });
  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({
        'aria-label': ARIA_LABEL,
        'aria-describedby': ARIA_DESCRIBE_BY,
      });
      const attributes = element.find('input').props();
      expect(attributes['aria-label']).toBe(ARIA_LABEL);
      expect(attributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, itemRef', () => {
      const TEXT_ID = 'testId';
      render({
        id: TEXT_ID,
        itemRef: 'ref',
      });
      const attributes = element.find('input').props();
      expect(attributes.id).toBe(TEXT_ID);
      expect(attributes.itemRef).toBe('ref');
    });
  });
});
