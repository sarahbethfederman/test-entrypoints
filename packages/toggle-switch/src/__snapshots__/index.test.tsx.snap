// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch should render the whole component 1`] = `
.c2 {
  position: absolute;
  opacity: 0;
  height: 0;
  width: 0;
}

.c2:focus ~ span {
  border: 2px solid #66d9c9;
  box-shadow: 0 0 3px 0 rgba(0,192,165,0.8);
}

.c1 {
  width: calc(2.5 * var(--lendi-ui-size));
  height: calc(2 * var(--lendi-ui-size));
  position: relative;
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 6px;
  box-sizing: border-box;
  margin-left: calc(0.5 * var(--lendi-ui-size));
}

.c3 {
  --width: calc(2.125 * var(--lendi-ui-size));
  --height: calc(0.875 * var(--lendi-ui-size));
  --handle-width: calc(1.25 * var(--lendi-ui-size));
  position: absolute;
  content: '';
  width: var(--width);
  height: var(--height);
  top: calc(50% - var(--height)/2);
  border-radius: 6px;
  margin-left: calc(var(--handle-width) * 0.3);
  margin-right: calc(var(--handle-width) * 0.3);
  box-sizing: border-box;
  background-color: #adadad;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c4 {
  --width: calc(1.25 * var(--lendi-ui-size));
  --track-width: calc(2.125 * var(--lendi-ui-size));
  --height: var(--width);
  --track-margin: var(--width) * 0.3;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.1), 0 2px 2px 0 rgba(0,0,0,0.12), 0 1px 3px 0 rgba(0,0,0,0.14);
  position: absolute;
  content: '';
  width: var(--width);
  height: var(--height);
  top: calc(50% - var(--height)/2);
  border: 1px solid transparent;
  background-color: #f8f8f8;
  border-radius: 50%;
  box-sizing: border-box;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  text-align: center;
}

.c6 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 6px;
  min-height: 48px;
  cursor: pointer;
}

.c0:hover > span > div {
  border-color: #00c0a5;
}

.c0:active > span > div {
  background-color: #777777;
}

.c5 {
  display: inline-block;
  margin-left: calc(0.75 * var(--lendi-ui-size));
}

<ToggleSwitch
  label="Test Toggle Switch"
  onChange={[MockFunction]}
  value="1"
>
  <styled.label
    checked={false}
    disabled={false}
    error={false}
    isBoxed={false}
  >
    <label
      checked={false}
      className="c0"
      disabled={false}
    >
      <Toggle
        isChecked={false}
        isDisabled={false}
        isError={false}
        isLoading={false}
        onChange={[MockFunction]}
        size="md"
        value="1"
      >
        <styled.span
          size="md"
        >
          <span
            className="c1"
            size="md"
          >
            <styled.input
              checked={false}
              disabled={false}
              error={false}
              onChange={[MockFunction]}
              type="checkbox"
              value="1"
            >
              <input
                checked={false}
                className="c2"
                disabled={false}
                onChange={[MockFunction]}
                type="checkbox"
                value="1"
              />
            </styled.input>
            <styled.span
              checked={false}
              disabled={false}
              error={false}
              size="md"
            >
              <span
                checked={false}
                className="c3"
                disabled={false}
                size="md"
              />
            </styled.span>
            <styled.div
              checked={false}
              disabled={false}
              error={false}
              size="md"
            >
              <div
                checked={false}
                className="c4"
                disabled={false}
                size="md"
              />
            </styled.div>
          </span>
        </styled.span>
      </Toggle>
      <Styled(styled.p)
        size="md"
      >
        <styled.p
          className="c5"
          size="md"
        >
          <p
            className="c5 c6"
            size="md"
          >
            Test Toggle Switch
          </p>
        </styled.p>
      </Styled(styled.p)>
    </label>
  </styled.label>
</ToggleSwitch>
`;
